Update Wallet logging

philipr-za opened the pull request:
## Description
This PR updates a number of logging statements levels in the wallet. Many error logs are brought down to warn if they are not related to an event that halts progress on the current task. Some info statements became debug and some debug were moved down to trace.

## How Has This Been Tested?
No functional changes.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
* [ ] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [x] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
* [x] I'm merging against the `development` branch.
* [x] I ran `cargo-fmt --all` before pushing.
* [ ] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

sdbondi commented on base_layer/wallet/src/transaction_service/protocols/transaction_receive_protocol.rs:
```diff
@@ -223,12 +223,15 @@ where TBackend: TransactionBackend + Clone + 'static
                             },
                             Err(e) => {
-                                error!(
+                                warn!(
```
Discovery failed: this could (should) generally be because the peer is offline. Since the fallback (store and forward) is in place, I'd say this could be debug level.

philipr-za replied:
fair enough

sdbondi commented on base_layer/wallet/src/transaction_service/protocols/transaction_receive_protocol.rs:
```diff
@@ -258,7 +261,7 @@ where TBackend: TransactionBackend + Clone + 'static
 
             if !direct_send_result && !store_and_forward_send_result {
-                warn!(
+                error!(
```
If this is retried again later, Id say warn! is correct?

philipr-za replied:
So this one terminates this protocol, it won't be retried later unless its done explicitly

philipr-za replied:
and right now there isn't actually an explicit way to do that so this is def an error because it shouldn't occur. If we received a Tx message then we should be able to send a reply via one of the routes.

sdbondi commented on base_layer/wallet/src/transaction_service/protocols/transaction_receive_protocol.rs:
```diff
@@ -269,9 +272,12 @@ where TBackend: TransactionBackend + Clone + 'static
             }
 
-            info!(
+            trace!(
```
Seems like good debug-level information ?

philipr-za replied:
I was on the fence about this one because there is a message just above it with the tx_id which is what you would need to trace thing. The amount and message is not really useful for debug.

sdbondi reviewed (commented):
LGTM - the log level to select is often more opinion than it ideally should be, but I think in any situation where the transaction should still go through it can be debug! rather than warn!.

philipr-za committed:
Update Wallet logging

This PR updates a number of logging statements levels in the wallet. Many error logs are brought down to warn if they are not related to an event that halts progress on the current task. Some info statements became debug and some debug were moved down to trace.
